AppHandling/Run-AlPipeline.ps1

```diff
@@ -1746,11 +1767,36 @@ Write-Host -ForegroundColor Yellow @'
         }
     }
 
+    # Script block for compiling an app using a compiler folder
+    $CompileAppWithBCCompilerFolder = {
+        Param(
+            [scriptblock] $PreCompileApp,
+            [scriptblock] $PostCompileApp,
+            $Parameters
+        )
+
+        if($PreCompileApp) {
+            Write-Host "Running custom pre-compilation script..."
+
+            Invoke-Command -ScriptBlock $PreCompileApp -ArgumentList $Parameters
+        }
+
+        $appFile = Compile-AppWithBcCompilerFolder @Parameters
+
+        if($PostCompileApp) {
+            Write-Host "Running custom post-compilation script..."
+
+            Invoke-Command -ScriptBlock $PostCompileApp -ArgumentList $Parameters
+        }
+        
+        return $appFile
+    }
+
     try {
         Write-Host "`nCompiling $($Parameters.appProjectFolder)"
         $withCops = $Parameters+$CopParameters
         if ($useCompilerFolder) {
-            $appFile = Compile-AppWithBcCompilerFolder @parameters
+            $appFile = Invoke-Command -ScriptBlock $CompileAppWithBCCompilerFolder -ArgumentList $PreCompileApp, $PostCompileApp, $Parameters

```

Comment:
We need to call the same pre and post when doing Compile-AppInBcContainer

Reply:
Why so? For `Compile-AppInBcContainer` there is already `CompileAppInBcContainer` override. It cannot be the same script (it should differ based on whether `useCompilerFolder` is on).

`PreCompileApp` and `PostCompileApp` are only expected when `useCompilerFolder` is on.

Reply:
Ok, my understanding was that the pre and post compile scripts are to cover the scenarios when `useCompilerFolder` is on and the custom `CompileAppInBcContainer` won't be called.

In any case, the only way one can use the same scripts for both when `useCompilerFolder` is on and off is to keep checking the value of the flag.
e.g.
```
if ($useCompilerFolder`) {
    # do something in the compiler folder
} 
else {
    # do something in the BC container
}
```